### ISSUE TITLE
Error handling code assumes response content is JSON

### ISSUE THREAD

**Issue by hwkns** [urbanairship/python-library#36]
When the API responds with `Error 503: Your request could not be completed, please retry.`, the call to `json.dumps(details)` raises an exception:
```
  File "/home/ubuntu/virtualenvs/venv-3.4.3/lib/python3.4/site-packages/urbanairship/push/core.py", line 49, in send
    version=3
  File "/home/ubuntu/virtualenvs/venv-3.4.3/lib/python3.4/site-packages/urbanairship/core.py", line 65, in _request
    raise common.AirshipFailure.from_response(response)
  File "/home/ubuntu/virtualenvs/venv-3.4.3/lib/python3.4/site-packages/urbanairship/common.py", line 75, in from_response
    response.status_code, error_code, error, json.dumps(details))
  File "/opt/circleci/python/3.4.3/lib/python3.4/json/__init__.py", line 230, in dumps
    return _default_encoder.encode(obj)
  File "/opt/circleci/python/3.4.3/lib/python3.4/json/encoder.py", line 192, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/opt/circleci/python/3.4.3/lib/python3.4/json/encoder.py", line 250, in iterencode
    return _iterencode(o, 0)
  File "/opt/circleci/python/3.4.3/lib/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
nose.proxy.TypeError: b'Error 503: Your request could not be completed, please retry.' is not JSON serializable
```

**Comment by rlepinski** [comment]
Thanks for bringing this to our attention. We will look into it for the next release.

**Comment by pdxmele** [comment]
This should now be fixed in 3.0.0
